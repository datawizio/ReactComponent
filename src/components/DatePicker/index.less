@import "~antd/es/datepicker/style/index";
@import "../../styles/variables";
