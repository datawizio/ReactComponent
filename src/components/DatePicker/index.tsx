import { Dayjs } from "dayjs";
import dayjsGenerateConfig from "rc-picker/lib/generate/dayjs";
import generatePicker from "antd/es/date-picker/generatePicker";
import "./index.less";

const DatePicker = generatePicker<Dayjs>(dayjsGenerateConfig);

export default DatePicker;
