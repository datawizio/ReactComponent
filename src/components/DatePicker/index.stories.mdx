import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import DatePicker from "./index";

<Meta title="Components/DatePicker/DatePicker" component={DatePicker} />

# DatePicker

Компонент `DatePicker`.

Базовый компонент:
[**antd/DatePicker**](https://ant.design/components/date-picker/).

## Basic

<Preview>
  <Story name="Basic">
    {() => {
      const handleChange = value => {
        {
          /* console.log(value.format("YYYY-MM-DD")); */
        }
      };
      return <DatePicker format="YYYY-MM-DD" onChange={handleChange} />;
    }}
  </Story>
</Preview>

## Props

- [**antd/DatePicker**](https://ant.design/components/date-picker/#API)

<Props of={DatePicker} />
