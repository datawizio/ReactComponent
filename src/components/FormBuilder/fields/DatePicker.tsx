import React, { useMemo } from "react";
import dayjs, { Dayjs } from "dayjs";
import { Form } from "antd";
import DatePicker, { CalendarTypes } from "../../DatePicker";
import { FieldDatePickerProps, FormFieldProps } from "../types";

interface FieldProps extends FormFieldProps<any> {
  format: string;
  storeFormat?: string;
  value: any;
  type?: CalendarTypes;
  disabledDate?: (currentDate: Dayjs) => boolean;
}

const Field: React.FC<FieldProps> = ({
  format,
  type,
  storeFormat,
  value,
  onChange,
  ...restProps
}) => {
  const formatedValue = useMemo<Dayjs>(() => {
    if (!value) return null;
    if (typeof value === "string") return dayjs(value);
    return value;
  }, [value]);
  const Component = DatePicker.Picker[type];
  return (
    //@ts-ignore
    <Component
      {...restProps}
      //@ts-ignore
      onChange={onChange}
      format={format}
      value={formatedValue}
    />
  );
};

Field.defaultProps = {
  type: "iso-8601"
};

export const FieldDatePicker: React.FC<FieldDatePickerProps> = React.memo(
  ({
    format,
    label,
    rules,
    name,
    placeholder,
    fullWidth,
    onChange,
    inputReadOnly,
    ...restProps
  }) => {
    const handleChange = (value: Dayjs) => {
      onChange && onChange({ name, value });
    };

    return (
      <Form.Item name={name} label={label} rules={rules}>
        <Field
          {...restProps}
          inputReadOnly={inputReadOnly}
          placeholder={placeholder}
          //@ts-ignore
          onChange={handleChange}
          format={format}
          className={fullWidth ? "ant-picker-w100" : ""}
        />
      </Form.Item>
    );
  }
);
