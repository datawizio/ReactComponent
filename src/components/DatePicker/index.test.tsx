import "jsdom-global/register";
import React from "react";
import { shallow } from "enzyme";

import DatePicker from "./index";

const mockProps = {};

const setUp = (props?) => shallow(<DatePicker {...props} />);

describe(" DatePicker component", () => {
  let component;
  beforeEach(() => {
    component = setUp(mockProps);
  });

  it("DatePicker rendered correctly", () => {
    expect(component).toMatchSnapshot();
  });
});
