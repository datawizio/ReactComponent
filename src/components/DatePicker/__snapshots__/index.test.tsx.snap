// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` DatePicker component DatePicker rendered correctly 1`] = `
<LocaleReceiver
  componentName="DatePicker"
  defaultLocale={[Function]}
>
  <Component />
</LocaleReceiver>
`;
